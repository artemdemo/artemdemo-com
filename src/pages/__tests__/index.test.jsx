import React from 'react';
import renderer from 'react-test-renderer';
import Index from '../index';

jest.mock('../../components/Container/Container');
jest.mock('../../components/Posts/Posts');
jest.mock('../../components/Pagination/Pagination');

describe('Index', () => {
    it('should render', () => {
        const tree = renderer.create(
            <Index
                data={{
                    site: {
                        siteMetadata: {
                            title: 'Some title',
                            description: 'Some description',
                        },
                    },
                    allMarkdownRemark: {
                        edges: [
                            {
                                frontmatter: {
                                    title: 'Some title',
                                    date: '10 May, 2018',
                                },
                                fields: {
                                    slug: '/first',
                                },
                                excerpt: 'Excerpt text',
                            }
                        ],
                    },
                }}
                location={{}}
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});