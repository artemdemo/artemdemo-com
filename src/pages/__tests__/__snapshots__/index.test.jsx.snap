// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index should render 1`] = `
<div
  data-mock="Container"
>
  <div
    data-mock="Helmet"
  >
    {
  "htmlAttributes": {
    "lang": "en"
  },
  "meta": [
    {
      "name": "description",
      "content": "Some description"
    }
  ],
  "title": "Some title"
}
  </div>
  <h3>
    <div
      data-mock="Link"
    >
      Some title
    </div>
  </h3>
  <p>
    Written by 
    <strong>
      Artem Demo
    </strong>
     who lives and works in Tel Aviv building useful things.
     
    <a
      href="https://github.com/artemdemo"
    >
      You should follow him on Github
    </a>
  </p>
  <div
    data-mock="Posts"
  >
    {
  "list": [
    {
      "frontmatter": {
        "title": "Some title",
        "date": "10 May, 2018"
      },
      "fields": {
        "slug": "/first"
      },
      "excerpt": "Excerpt text"
    }
  ]
}
  </div>
</div>
`;
